servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1783,62 +1795,66 @@ private void setStateFromEntry(SetStateEntry entry) throws FileDoesNotExistExcep
   }
 
   /**
-   * Checks user's permission on a path.
+   * Checks user's permission on a path. If the path is invalid, it should bypass the
+   * {@link InvalidPathException} and the logic at operation will handle it.
    *
    * @param action requested {@link FileSystemAction} by user
    * @param path the path to check permission on
    * @param checkParent indicates whether to check its parent
    * @throws AccessControlException if permission checking fails
-   * @throws InvalidPathException if the path is invalid
    */
-  private void checkPermission(FileSystemAction action, TachyonURI path,
-      boolean checkParent) throws AccessControlException, InvalidPathException {
+  private void checkPermission(FileSystemAction action, TachyonURI path, boolean checkParent)
+      throws AccessControlException {
     // bypasses permission checking if security is not enabled.
     if (!SecurityUtils.isSecurityEnabled(MasterContext.getConf())) {
       return;
     }
 
-    // collects inodes info on the path
-    List<Inode> inodes = mInodeTree.collectInodes(path);
-    List<FileInfo> fileInfos = new ArrayList<FileInfo>();
-    for (Inode inode : inodes) {
-      fileInfos.add(inode.generateClientFileInfo(mInodeTree.getPath(inode).toString()));
-    }
+    try {
+      // collects inodes info on the path
+      List<Inode> inodes = mInodeTree.collectInodes(path);
+      List<FileInfo> fileInfos = new ArrayList<FileInfo>();
+      for (Inode inode : inodes) {
+        fileInfos.add(inode.generateClientFileInfo(mInodeTree.getPath(inode).toString()));
+      }
 
-    String[] pathComponents = PathUtils.getPathComponents(path.getPath());
-    if (pathComponents.length < fileInfos.size()) {
-      throw new InvalidPathException(ExceptionMessage.PATH_INVALID.getMessage(path.getPath()));
-    }
+      String[] pathComponents = PathUtils.getPathComponents(path.getPath());
+      if (pathComponents.length < fileInfos.size()) {
```

Comment:
Normally `pathComponents.length` is not possible to be greater than `fileInfos.size()`. Just add this for full coverage of all case. So it can be removed.